storage/database/db_manager.go

```diff
@@ -2027,3 +2031,22 @@ func (dbm *databaseManager) ReadGovernanceState() ([]byte, error) {
 	db := dbm.getDatabase(MiscDB)
 	return db.Get(governanceStateKey)
 }
+
+func (dbm *databaseManager) WriteChainDataFetcherCheckpoint(checkpoint uint64) error {
+	db := dbm.getDatabase(MiscDB)
+	return db.Put(chaindatafetcherCheckpointKey, common.Int64ToByteBigEndian(checkpoint))
+}
+
+func (dbm *databaseManager) ReadChainDataFetcherCheckpoint() (uint64, error) {
+	db := dbm.getDatabase(MiscDB)
+	hasCheckpoint, err := db.Has(chaindatafetcherCheckpointKey)
+	if err != nil || !hasCheckpoint {
+		return 0, err
+	}
+
+	data, err := db.Get(chaindatafetcherCheckpointKey)
+	if err != nil {
+		return 0, err
+	}
```

Comment:
AFAIK... In most DB, `Has` calls `Get` method internally and then checks its result. 
How about just calling `Get` method only and checking its returns (both data and err)?

Reply:
Right, but it was hard to handle the error when key does not exist depending on the database.
This is because the errors are different based on them.

Reply:
I think the current code also cannot distinguish the existence of the item for all databases. 
If you need to distinguish the difference, you better define new errors in this area. Currently, some database return  `0, nil` when the item doesn't exist, but another database returns `0, error`.

Reply:
Right... The `Has` method works differently based on the sort of db. I guess I need to handle the existence error db by db.

Reply:
Ok. Let's go back to the original subject. Do we still need to use `Has` method? Each of its returns needs to be handled just like `Get`.

Reply:
It is not needed if the error returned by `Get` method is distinguishable. Let me check about this.

Reply:
Please take another look.